File: templates/worker/ts/workflow.ts
/**
 * Render Workflow
 *
 * Uses the Render SDK to orchestrate multi-step workflows.
 * See: https://github.com/render-oss/sdk
 */

import "dotenv/config";
import Render from "@renderinc/sdk";

// Initialize Render SDK
const render = new Render({
  apiKey: process.env.RENDER_API_KEY,
});

/**
 * Example workflow: Deploy and notify
 */
async function deployWorkflow(): Promise<void> {
  console.log("Starting deployment workflow...");

  // TODO: Customize your workflow steps
  // The Render SDK provides access to:
  // - Services management
  // - Deployments
  // - Environment variables
  // - Logs and metrics

  try {
    // Example: List services
    const services = await render.services.list();
    console.log(`Found ${services.length} services`);

    // Example: Trigger a deploy (uncomment and customize)
    // const deployment = await render.deploys.create({
    //   serviceId: 'srv-xxxxx',
    // });
    // console.log(`Deployment triggered: ${deployment.id}`);

    console.log("Workflow completed successfully");
  } catch (error) {
    console.error("Workflow failed:", error);
    throw error;
  }
}

/**
 * Main entry point
 */
async function main(): Promise<void> {
  if (!process.env.RENDER_API_KEY) {
    console.warn("RENDER_API_KEY not set - workflow will run in demo mode");
  }

  await deployWorkflow();
}

// Run the workflow
main()
  .then(() => {
    console.log("Workflow exiting successfully");
    process.exit(0);
  })
  .catch((error) => {
    console.error("Workflow exiting with error:", error);
    process.exit(1);
  });

File: templates/worker/py/workflow.py
"""
Render Workflow

Uses the Render SDK to orchestrate multi-step workflows.
See: https://pypi.org/project/render-sdk/
"""

import os
import sys

from dotenv import load_dotenv

load_dotenv()

# Check for render-sdk availability
try:
    import render
except ImportError:
    print("render-sdk not installed. Run: pip install render-sdk")
    sys.exit(1)


def deploy_workflow() -> None:
    """Example workflow: Deploy and notify."""
    print("Starting deployment workflow...")

    api_key = os.getenv("RENDER_API_KEY")

    if not api_key:
        print("RENDER_API_KEY not set - workflow will run in demo mode")
        print("Workflow completed (demo mode)")
        return

    # Initialize Render client
    client = render.Render(api_key=api_key)

    try:
        # TODO: Customize your workflow steps
        # The Render SDK provides access to:
        # - Services management
        # - Deployments
        # - Environment variables
        # - Logs and metrics

        # Example: List services
        services = client.services.list()
        print(f"Found {len(services)} services")

        # Example: Trigger a deploy (uncomment and customize)
        # deployment = client.deploys.create(service_id="srv-xxxxx")
        # print(f"Deployment triggered: {deployment.id}")

        print("Workflow completed successfully")

    except Exception as e:
        print(f"Workflow failed: {e}")
        raise


def main() -> None:
    """Main entry point."""
    deploy_workflow()


if __name__ == "__main__":
    try:
        main()
        print("Workflow exiting successfully")
        sys.exit(0)
    except Exception as e:
        print(f"Workflow exiting with error: {e}")
        sys.exit(1)
